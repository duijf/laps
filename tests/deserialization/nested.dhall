let Laps = ../../package.dhall

let mkUnit
    : Text -> Laps.Unit
    =     \(name : Text)
      ->  { alias = name
          , executable =
              Laps.program { program = "foo", arguments = [ "--bar", "--baz" ] }
          , nixEnv = None Laps.NixEnv
          , watchExtensions = [] : List Text
          }

let mkSingle
    : Text -> Laps.StartOrder
    = \(name : Text) -> Laps.single (mkUnit name)

in  [ { name = "foo"
      , shortDesc = "foo"
      , startOrder =
          Laps.tree
            (mkUnit "foo")
            [ mkSingle "bar"
            , Laps.parallel [ mkSingle "baz", mkSingle "quix" ]
            , Laps.serial
                [ mkSingle "1"
                , mkSingle "2"
                , Laps.tree (mkUnit "3") [ mkSingle "4" ]
                ]
            ]
      }
    ]
