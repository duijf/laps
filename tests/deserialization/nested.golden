[ Command
    { name = "foo"
    , shortDesc = "foo"
    , startOrder = Tree
        ( Unit
            { uExecutable = P
                ( Program
                    { program = "foo"
                    , arguments =
                        [ "--bar"
                        , "--baz"
                        ]
                    }
                )
            , uAlias = "foo"
            , uNixEnv = Nothing
            , uWatchExtensions = []
            }
        )
        [ Single
            ( Unit
                { uExecutable = P
                    ( Program
                        { program = "foo"
                        , arguments =
                            [ "--bar"
                            , "--baz"
                            ]
                        }
                    )
                , uAlias = "bar"
                , uNixEnv = Nothing
                , uWatchExtensions = []
                }
            )
        , Parallel
            [ Single
                ( Unit
                    { uExecutable = P
                        ( Program
                            { program = "foo"
                            , arguments =
                                [ "--bar"
                                , "--baz"
                                ]
                            }
                        )
                    , uAlias = "baz"
                    , uNixEnv = Nothing
                    , uWatchExtensions = []
                    }
                )
            , Single
                ( Unit
                    { uExecutable = P
                        ( Program
                            { program = "foo"
                            , arguments =
                                [ "--bar"
                                , "--baz"
                                ]
                            }
                        )
                    , uAlias = "quix"
                    , uNixEnv = Nothing
                    , uWatchExtensions = []
                    }
                )
            ]
        , Serial
            [ Single
                ( Unit
                    { uExecutable = P
                        ( Program
                            { program = "foo"
                            , arguments =
                                [ "--bar"
                                , "--baz"
                                ]
                            }
                        )
                    , uAlias = "1"
                    , uNixEnv = Nothing
                    , uWatchExtensions = []
                    }
                )
            , Single
                ( Unit
                    { uExecutable = P
                        ( Program
                            { program = "foo"
                            , arguments =
                                [ "--bar"
                                , "--baz"
                                ]
                            }
                        )
                    , uAlias = "2"
                    , uNixEnv = Nothing
                    , uWatchExtensions = []
                    }
                )
            , Tree
                ( Unit
                    { uExecutable = P
                        ( Program
                            { program = "foo"
                            , arguments =
                                [ "--bar"
                                , "--baz"
                                ]
                            }
                        )
                    , uAlias = "3"
                    , uNixEnv = Nothing
                    , uWatchExtensions = []
                    }
                )
                [ Single
                    ( Unit
                        { uExecutable = P
                            ( Program
                                { program = "foo"
                                , arguments =
                                    [ "--bar"
                                    , "--baz"
                                    ]
                                }
                            )
                        , uAlias = "4"
                        , uNixEnv = Nothing
                        , uWatchExtensions = []
                        }
                    )
                ]
            ]
        ]
    }
]